lib/Onnxifi/Base.cpp

```diff
@@ -135,7 +136,48 @@ onnxStatus Graph::setIOAndRun(uint32_t inputsCount,
     ctx->getPlaceholderBindings()->insert(outPhPtr, std::move(t));
   }
 
-  return run(std::move(ctx), outputEvent, std::move(phNameToOnnxTensorOutputs));
+  return run(std::move(ctx), outputEvent, std::move(phNameToOnnxTensorOutputs),
+             traceEvents);
+}
+
+void Graph::setTraceEvents(onnxTraceEventList *traceEvents,
+                           const TraceContext &traceContext) {
+  if (!traceEvents) {
+    return;
+  }
+
+  std::vector<onnxTraceEvent *> traceEventsVec;
+  for (const auto &glowTraceEvent : traceContext.getTraceEvents()) {
+    auto *traceEvent = new onnxTraceEvent();
+    assert(
+        glowTraceEvent.type.size() == 1 &&
+        "Events with types longer than a single char not supported by onnxifi");
+    traceEvent->eventType = glowTraceEvent.type[0];
+    traceEvent->timestamp = glowTraceEvent.timestamp;
+    traceEvent->tid = glowTraceEvent.tid;
+    char *eventName = new char[glowTraceEvent.name.size() + 1];
```

Comment:
why use a temporary here rather than just allocating into `traceEvent->eventName`?

Reply:
cause it's a const*